Pdu.hpp

```diff
@@ -0,0 +1,59 @@
+#pragma once
+#include <cstdint>
+#include<string>
+#include <bitset>
+
+class Pdu
+{
+private:
+	static constexpr int SIZE_BYTES = 5;
+	static constexpr int NAME_SIZE = 30;
+	static constexpr int MESSAGE_SIZE = 1024;
+	uint32_t _size;
+	char _name[NAME_SIZE];
+	char _message[MESSAGE_SIZE];
+	char* _buffer;
+	
+public:
+	Pdu();
+	Pdu(uint32_t size,const char* name,const char* message);
+	~Pdu();
+	/**
+ * @brief Creates a PDU from the current object's data.
+ *
+ * Allocates memory for the PDU buffer and populates it with the size, name, and
+ * message in a specific format. The buffer includes the size, name, and message
+ * fields sequentially.
+ *
+ * @return A pointer to the internal buffer containing the PDU data.
+ */
+	const char* writePdu();
```

Comment:
there is a lot to unpack here. other than using `std::string` or any other c++ type
how do you know the buffer's length?